modelling/src/main/java/org/axonframework/modelling/command/AnnotationCommandTargetResolver.java

```diff
@@ -44,6 +46,35 @@
  */
 public class AnnotationCommandTargetResolver implements CommandTargetResolver {
 
+    private final Class<? extends Annotation> identifierAnnotation;
+    private final Class<? extends Annotation> versionAnnotation;
+
+    /**
+     * Instantiate a Builder to be able to create a {@link AnnotationCommandTargetResolver}.
+     * <p>
+     * The TargetAggregateIdentifierAnnotation is defaulted to {@link TargetAggregateIdentifier}, 
+     * TargetAggregateVersionAnnotation to {@link TargetAggregateVersion}.
+     *
+     * @return a Builder to be able to create a{@link AnnotationCommandTargetResolver}
+     */
+    public static final Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * @deprecated Please use the {@link #builder()}.
+     */
+    @Deprecated
+    public AnnotationCommandTargetResolver() {
+        this.identifierAnnotation = TargetAggregateIdentifier.class;
+        this.versionAnnotation = TargetAggregateVersion.class;
+    }
+
+    protected AnnotationCommandTargetResolver(Builder builder) {
```

Comment:
Nit: People might override this constructor. As such, it's visible, so I would expect it to have some javadoc.

Reply:
Done. Please consider/discuss, if it is really necessary to document everything with JavaDoc. Of course, public API methods need it, no question about that. But i'm not sure about e.g. the "build"-Method in an Builder. I'm guilty of overdoing javadoc in my company too, because it get measured by SONAR. But if you watch e.g. Adam Bien videos, he does not say too many nice things about JavaDoc because of that.

Reply:
Thanks for the suggestion @JohT, we'll take note of that.
Reasoning for having it here is, indeed, because it's public API, which we'd like to be as explanatory as possible, without requiring users to actually read the code.